src/object/cli_obj.c

```diff
@@ -2392,17 +2399,43 @@ obj_update_csums(const struct dc_object *obj, const daos_obj_update_t *args) {
 			((char *) iod->iod_csums->cs_csum)[0]++;
 	}
 }
+static int
+obj_retry_csum_err(struct dc_object *obj, struct obj_auxi_args *obj_auxi,
+		     uint64_t dkey_hash, unsigned int map_ver, uint8_t *bitmap)
+{
+	unsigned int tgt_shard, shard_cnt, shard_idx;
+	int rc = 0;
+
+
+	rc = obj_dkey2grpmemb(obj, dkey_hash, map_ver,
+			      &shard_idx, &shard_cnt);
+	if (rc != 0)
+		goto out;
+
+	if (shard_cnt < 2) {
+		rc = -DER_CSUM;
+		goto out;
+	}
+	tgt_shard = (obj_auxi->req_tgts.ort_shard_tgts[0].st_shard + 1)
```

Comment:
if all targets are corrupt, this will just loop through targets infinitely?

Reply:
Yes, but I don't want to add state to the client to track the retries.

Reply:
I could change it to not move past the last replica in the group.  Right now, fetch always goes to the leader (the first replica), so this would stop the loop after all replicas have been checked.  However, we intend to change fetch to "load balance" across the replicas, so once this change is made, this simple method will not check all of the replicas.